dashboard/app/controllers/script_levels_controller.rb

```diff
@@ -127,6 +127,15 @@ def toggle_hidden
     render json: []
   end
 
+  def instructions
+    require_levelbuilder_mode
+    authorize! :read, ScriptLevel
+
+    script = Script.get_from_cache(params[:script_id])
+
+    render 'levels/instructions', locals: { stages: script.stages }
```

Comment:
I'm not sure — this convention existed before I started working on this codebase.  For deeply-nested views it's a little easier than passing locals through each partial.  You're right that `local_assigns` are probably better.

One small nit: you don't need `stages = local_assigns[:stages]` (you can access `stages` directly).